aiida/backends/sqlalchemy/models/log.py

```diff
@@ -29,36 +29,48 @@ class DbLog(Base):
 
     id = Column(Integer, primary_key=True)
 
-    uuid = Column(UUID(as_uuid=True), default=uuid_func)
+    uuid = Column(UUID(as_uuid=True), default=uuid_func, unique=True)
 
     time = Column(DateTime(timezone=True), default=timezone.now)
     loggername = Column(String(255), index=True)
     levelname = Column(String(255), index=True)
 
-    objname = Column(String(255), index=True)
-    objuuid = Column(UUID(as_uuid=True), default=uuid_func, unique=False, index=True)
+    dbnode_id = Column(
+        Integer,
+        ForeignKey('db_dbnode.id', deferrable=True, initially="DEFERRED", ondelete="CASCADE"),
+        nullable=False
+    )
 
     message = Column(Text(), nullable=True)
     _metadata = Column('metadata', JSONB)
 
-    def __init__(self, time, uuid=None, loggername="", levelname="", objname="", objuuid=None,
-                 message=None, metadata=None):
+    ######### RELATIONSSHIP #########
+    dbnode = relationship(
+        'DbNode',
+        backref=backref('dblogs', passive_deletes='all', cascade='merge')
+    )
+
+    def __init__(self, time, uuid=None, loggername="", levelname="", dbnode_id=None,
+                 message=None, metadata=None, **kwargs):
 
         if not loggername or not levelname:
             raise ValidationError(
                 "The loggername and levelname can't be empty")
 
+        _metadata = kwargs['_metadata'] if '_metadata' in kwargs else None
+
+        if metadata is not None and _metadata is not None:
```

Comment:
Wouldn't it be better to demand it only be passed in the `kwargs` or directly as `metadata`? Who is calling this constructor, only us correct? Think it might be better to make sure of this from the caller side.